### PR TITLE
CEXT-6387: Extend per-item nested ACL support to the remaining Admin UI SDK components (grid columns, order view buttons, mass actions)

### PR DESCRIPTION
**Pull request by oshmyheliuk**
## Description

Added aclProtected as a documented optional field across usage.md, the app-management skill reference tables, and the app-migration skill mapping tables for all four extension point types (mass actions, grid
  columns, order view buttons, menu). Tightened GridColumnSchema from v.object to v.strictObject to reject unknown keys, and extended unit tests for getMassActionAclResourceId and getOrderViewButtonAclResourceId
  with blank and whitespace-only edge cases.

https://jira.corp.adobe.com/browse/CEXT-6387

## Related PR

https://github.com/magento-commerce/adobe-commerce-backend-uix/pull/356

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have read the **DEVELOPMENT** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/aio-commerce-lib-admin-ui/source/api/lib/acl-resource-id.ts
Comment (obarcelonap) [inline packages/aio-commerce-lib-admin-ui/source/api/lib/acl-resource-id.ts:1]: `aio-commerce-lib-admin-ui` is splitted by domain. This file is splitting by component. In order to get consistent user experience we should split by domain here too, meaning putting each function under its own module (menu, grid-columns, mass-actions..).

```typescript
// Incosistent
import { parseGridRequest } from "@adobe/aio-commerce-lib-admin-ui/grid-columns";
import { getGridColumnAclResourceId } from "@adobe/aio-commerce-lib-admin-ui/api";

// Consistent
import { getGridColumnAclResourceId, parseGridRequest } from "@adobe/aio-commerce-lib-admin-ui/grid-columns";
```

File: packages/aio-commerce-lib-admin-ui/docs/usage.md [hunk 1]
```diff
@@ -375,3 +375,30 @@ const allowed = await permissionClient.check(
   getMenuAclResourceId("approval-dashboard-app", "approval_dashboard"),
 );
 ```
+
+#### Resource ids for grid columns, view buttons, and mass actions
```
Comment (obarcelonap) [inline packages/aio-commerce-lib-admin-ui/docs/usage.md:377]: could you add an example on how to use this in a runtime action in conjuction with the other utilities of this package (request/response builders,...)

File: packages/aio-commerce-lib-admin-ui/source/api/lib/acl-resource-id.ts [hunk 2]
```diff
@@ -59,3 +59,72 @@ export function getMenuAclResourceId(
...
+export type AdminUiEntity = "order" | "product" | "customer";
+
+/**
+ * Derives the deterministic Commerce ACL resource id for a grid column.
+ *
+ * @param metadataId - The application's `metadata.id` value (e.g. `"approval-dashboard-app"`).
+ * @param entity - The grid's Commerce entity (`"order"`, `"product"`, or `"customer"`).
+ * @param columnId - The column's `id` value from `adminUi.<entity>.gridColumns.columns[].id`.
+ * @returns The full Commerce ACL resource id for the grid-column leaf node (e.g.
+ *   `"Magento_CommerceBackendUix::adminuisdk_app_approval_dashboard_app_order_gridcolumns_order_status"`),
```
Comment (obarcelonap) [inline packages/aio-commerce-lib-admin-ui/source/api/lib/acl-resource-id.ts]: Is `Magento_CommerceBackendUix::adminuisdk_app_` a constant value or is just an example? Maybe we can document this because for is no obvious how the example was built from the given args.
This comment applies to all the functions of this file

**Review by obarcelonap — APPROVED** [review]
LGTM but try to attend my comments :)

**Review by jcuerdo — APPROVED** [review]

File: packages/aio-commerce-lib-admin-ui/source/api/lib/acl-resource-id.ts
Comment (oshmyheliuk) [reply packages/aio-commerce-lib-admin-ui/source/api/lib/acl-resource-id.ts:1]: Moved each function to the related domain.

File: packages/aio-commerce-lib-admin-ui/docs/usage.md (hunk 1 above)
Comment (oshmyheliuk) [reply packages/aio-commerce-lib-admin-ui/docs/usage.md:377]: Added example

File: packages/aio-commerce-lib-admin-ui/source/api/lib/acl-resource-id.ts (hunk 2 above)
Comment (oshmyheliuk) [reply packages/aio-commerce-lib-admin-ui/source/api/lib/acl-resource-id.ts]: `Magento_CommerceBackendUix::adminuisdk_app_` this is a constant. This prefix will always append to the generated resource ID

Comment (oshmyheliuk) [reply packages/aio-commerce-lib-admin-ui/source/api/lib/acl-resource-id.ts]: Added descriptions to the constant and each component
